src/Pages/ControlPanel.jsx

```diff
@@ -25,24 +25,46 @@ const ControlPanel = () => {
 	const isSmall = useMediaQuery(theme.breakpoints.down("md"));
 
 	const [ctrlVals, dispatchCtrlVals] = useReducer(ctrlValsReducer, { remote: null, local: {} });
-	const fetchRemoteValues = (discardLocal) => {
-		dispatchCtrlVals({ type: "discard_remote" });
-		Promise.all(
-			systemControlsCollections.map(collName => getDocs(collection(db, collName)))
-		).then(querySnapshotArray => {
-			// get an array of arrays of [path, data] pairs, then flatten it to a array of [path, data] pairs
-			const pairs = querySnapshotArray.map(querySnapshot =>
-				// get an array of [path, data] pairs
-				querySnapshot.docs.map(doc => [doc.ref.path, doc.data()])
-			).flat();
-			if (discardLocal) {
-				// do this first to avoid the unnecessary checks in replace_remote
-				dispatchCtrlVals({ type: "discard_local" });
-			}
-			dispatchCtrlVals({ type: "replace_remote", newRemote: Object.fromEntries(pairs) });
-		});
-	};
-	useEffect(() => fetchRemoteValues(), []);
+
+  const fetchRemoteValues = (discardLocal) => {
+    useEffect(() => {
+      dispatchCtrlVals({ type: "discard_remote" });
+
+      
+      //this is the listener for the realtime updates
+      systemControlsCollections.map((collName) => {
+        const collectionRef = query(collection(db, collName));
+        console.log(collName)
+          const unsubscribe = onSnapshot(collectionRef, (querySnapshot) => {
+            const pairs = querySnapshot.docs.map(doc => [doc.ref.path, doc.data()]);
+            if (discardLocal) {
```

Comment:
Actually now that I think about it, it might be useful if we want an realtime update on the control panel to override whatever local changes the current user made. Though I'm not sure why we'd want that.... You can either delete it or just set it to false when the function is actually called; up to you.

Reply:
it's not used for real-time updates purposes right now, although u are right it could be. But fetchRemoteValues is also called when you make local changes in the ControlsOverviewPanel component. So i'm gonna leave it in and set it to false initially.